core/src/crawl_client.rs

```diff
@@ -82,19 +83,35 @@ pub async fn run(
 		..
 	}) = message_rx.recv().await
 	{
-		let block = match types::BlockVerified::try_from((header, None)) {
+		let block = match types::BlockVerified::try_from((header.clone(), None)) {
 			Ok(block) => block,
 			Err(error) => {
 				error!("Header is not valid: {error}");
 				continue;
 			},
 		};
-
 		let Some(extension) = &block.extension else {
 			info!("Skipping block without header extension");
 			continue;
 		};
 
+		let Some((rows, cols, _, _)) = extract_kate(&header.extension) else {
+			info!("Skipping block without header extension");
+			continue;
+		};
+		let Some(dimensions) = Dimensions::new(rows, cols) else {
+			info!(
+				block.block_num,
+				"Skipping block with invalid dimensions {rows}x{cols}",
+			);
+			continue;
+		};
+
+		if dimensions.cols().get() <= 2 {
+			error!(block.block_num, "More than 2 columns are required");
+			continue;
+		}
+
```

Comment:
`BlockVerified::try_from` is already doing "extraction", can we extend this type (e.g. with dimensions field), instead of repeating the same code?